mtp_api/apps/notification/rules.py

```diff
@@ -0,0 +1,273 @@
+from functools import reduce
+
+from django.db.models import Q
+from django.utils.translation import gettext_lazy as _
+from extended_choices import Choices
+
+from credit.models import Credit
+from disbursement.constants import DISBURSEMENT_RESOLUTION
+from disbursement.models import Disbursement
+from mtp_auth.models import PrisonUserMapping
+from security.constants import TIME_PERIOD
+from security.models import PrisonerProfile, SenderProfile, RecipientProfile
+from .models import Event, EventCredit, EventDisbursement
+
+INPUT_TYPES = Choices(
+    ('NUMBER', 'number', _('Number')),
+    ('STRING', 'string', _('String')),
+)
+
+
+class BaseRule():
```

Comment:
does this stop it being a subclass of object? probs best without `()`

Reply:
No, but sure.